File: js/Anagram.js
function isAnagram(s,t)
{
    if(s.length!=t.length)
    {
        return false
    }
    let seen={}
    for(let ch of s)
    {
        if(seen[ch])
        {
            seen[ch]++
        }
        else
        {
            seen[ch]=1

        }
    }
    for(let ch of t)
    {
        if(!ch in seen)
        {
            return false
        }
        seen[ch]--
    }
    if(seen[ch]<0)
    {
        return false
    }
    return true
}

console.log(isAnagram("anagram", "nagaram"))
console.log(isAnagram("rat", "car"))         

File: anagram.py
class Solution:
    def anagram(self,s,t):
        if len(s)!=len(t):
            return False
        count={}
        for char in s:
            if char in count:
                count[char]+=1
            else:
                count[char]=1
        for ch in t:
            if ch not in count:
                return False
            count[ch]-=1 
            if count[ch] <0:
                return False
        return True
s=Solution()
print(s.anagram("anagram", "nagaram"))
print(s.anagram("rat", "car"))                   